File: Servicelane.java
import java.io.*;
import java.util.*;
public class Servicelane
{
    public static void main(String args[])
    {
        int N,T,i;
        Scanner sc=new Scanner(System.in);
        N=sc.nextInt();
        T=sc.nextInt();
        sc.nextLine();
        int width[]=new int[N];
        
        for(i=0;i<N;i++)
        {
            width[i]=sc.nextInt();
        }
        while(T-->0)
        {
            int start,end;
            start=sc.nextInt();
            end=sc.nextInt();
            int min=Integer.MAX_VALUE;
            for(i=start;i<=end;i++)
            {
                if(width[i]<min)
                   min=width[i];
            }
            
            System.out.println(min);
            
        }
        
    }
}

File: Servicelane.cpp
#include<iostream>
#include<climits>
using namespace std;
int width[100000];
int main()
{
    int w,t,i=0;
    cin>>w>>t;
    while(i<w)
    {
        cin>>width[i];
        i++;
    }
    
    while(t--)
    {
    int start,end,min=INT_MAX;
    cin>>start>>end;
    for(int j=start;j<=end;j++)
    {
        if(width[j]<min)
        {
            min=width[j];    
        }
       
    }
    
    cout<<min<<endl;
    }
    return 0;
}
